### PR TITLE
feat(ia-listings): IA for Listings Plugin

### PR DESCRIPTION
**Pull request by ronchambers**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This code completes Asana Task: "Newspack Information Architecture > IA Listings".  

Screenshot of Menu:

ia-listings-changes

* Menu position and icon updated.
* All sub menu items have the new Admin Header.
* Links to categories/tags removed.
* Link for Marketplace moved above Generic ( SEE RELATED: https://github.com/Automattic/newspack-listings/pull/487 )

Screenshot of Dashboard:

ia-listings-dashboard

* "Cards" are clickable to Listings menu items.
* When Listings Plugin is inactive, entire section and cards are removed.

### How to test the changes in this Pull Request:

1. Pull branch.
2. `npm run build`
3. Activate Newspack plugin.
4. Install/activate Newspack Listings Plugin.
5. Verify screenshots and notes above.
6. Deactivate Listings plugin, verify dashboard no longer shows Listings section.

### Note about Related PR:

The code to move the submenu link for Marketplace above Generic is done in the Listings Plugin. ( SEE RELATED: https://github.com/Automattic/newspack-listings/pull/487 ) 

If we are unable to approve that PR, then we'll need to approve this PR keeping Generic above Marketplace (this is how Listings Plugin does it already), or we'll need to update this PR with a solution.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Comment by ronchambers** [comment]
@miguelpeixe - please see above. I'll assign you in Asana and reach out in Slack.

File: includes/wizards/class-listings-wizard.php [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+	 * @var string
+	 */
+	protected $slug = 'newspack-listings';
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		
+		if ( ! is_plugin_active( 'newspack-listings/newspack-listings.php' ) ) {
```
Comment (miguelpeixe) [inline includes/wizards/class-listings-wizard.php]:
```suggestion
		if ( ! defined( 'NEWSPACK_LISTINGS_FILE' ) ) {
```

The path check is a bit fragile. It's not uncommon to use a different path when using a custom version for testing or other reasons.

File: includes/wizards/newspack/class-newspack-dashboard.php [hunk 2]
```diff
@@ -129,7 +129,11 @@ public function get_dashboard() {
 					],
 				],
 			],
-			'listings'             => [
+		];
+
+		// Newspack Listings Plugin.
+		if ( is_plugin_active( 'newspack-listings/newspack-listings.php' ) ) {
```
Comment (miguelpeixe) [inline includes/wizards/newspack/class-newspack-dashboard.php]:
```suggestion
		if ( defined( 'NEWSPACK_LISTINGS_FILE' ) ) {
```

Same as the other comment.

**Review by miguelpeixe — CHANGES_REQUESTED** [review]
It tested well! I have just a minor request inline.

File: includes/wizards/class-listings-wizard.php (hunk 1 above)
Comment (ronchambers) [reply includes/wizards/class-listings-wizard.php]: Agreed!

File: includes/wizards/newspack/class-newspack-dashboard.php (hunk 2 above)
Comment (ronchambers) [reply includes/wizards/newspack/class-newspack-dashboard.php]: Agreed!

**Comment by miguelpeixe** [comment]
@jaredrethman I've pushed some changes to accommodate the recent refactor to the menu order logic. Could you review this one?

**Review by jaredrethman — CHANGES_REQUESTED** [review]
@miguelpeixe thanks for the review, only thing missing is active submenu when add/edit post types i.e:
`/wp-admin/post-new.php?post_type=newspack_lst_*`
`/wp-admin/post.php?post=POST_ID`

**Comment by miguelpeixe** [comment]
Thank you for catching that, @jaredrethman!

Fixed in e6fbff0694a0457f54fcb6aa6b3bfbace71275a3

**Review by jaredrethman — APPROVED** [review]
@miguelpeixe thanks for the updates - code looks good - approved!

**Pull request merged**